lib/checkassert.cpp

```diff
@@ -120,37 +100,174 @@ void CheckAssert::assignmentInAssertError(const Token *tok, const std::string& v
                 "builds, this is a bug.", CWE398, Certainty::normal);
 }
 
-// checks if side effects happen on the variable prior to tmp
-void CheckAssert::checkVariableAssignment(const Token* assignTok, const Scope *assertionScope)
-{
-    if (!assignTok->isAssignmentOp() && assignTok->tokType() != Token::eIncDecOp)
-        return;
+//---------------------------------------------------------------------------
+
+/**
+ * \brief checks if a variable is assigned
+ * \retval true   - is variable assignment
```

Comment:
I don't think this is how you write a proper doxygen comment. I believe the `\retval` should only be written once.

Reply:
The name is a bit misleading. there is no check that some variable is assigned. It can be assignment of hardware address or assign some byte in a buffer etc..
The name `isAssignment` is a bit misleading also but I think it is better.